src/common/file.cpp

```diff
@@ -291,6 +300,14 @@ namespace tools {
       return true;
     }
 
+    fs::path depreciated_path = get_depreciated_default_data_dir();
+    if (fs::is_directory(depreciated_path, ec))
+    {
```

Comment:
I don't think this is the right place for this rename+symlink -- this `create_directories_if_necessary` is meant to be a generic "create this directory path" call (basically what `mkdir -p` does), but we only want to do this for the specific default data dir.  I suggest moving this just before the call in daemon/main.cpp.